File: 프로그래머스/2/131704. 택배상자/택배상자.java
import java.util.*;
import java.util.stream.*;

class Solution {
    public int solution(int[] order) {
        Deque<Integer> stack = new LinkedList<>();
        Queue<Integer> belt = IntStream.rangeClosed(1, order.length).boxed().collect(Collectors.toCollection(LinkedList::new));
        
        int i = 0;
        
        while (i < order.length) {
            int o = order[i];
            if (!belt.isEmpty() && belt.peek() == o) {
                belt.poll();
                i++;
            } else if (!stack.isEmpty() && stack.peekLast() == o) {
                stack.pollLast();
                i++;
            } else {
                if (!belt.isEmpty()) {
                    stack.add(belt.poll());
                } else {
                    break;
                }
            }
        }
        
        return i;
        
    }
}

File: 프로그래머스/2/131704. 택배상자/택배상자.py
from collections import deque

def solution(order):
    q = deque([i for i in range(1, len(order) + 1)])
    bozo_q = deque()
    
    answer = 0 
    for i in range(len(order)):
        if bozo_q and bozo_q[-1] == order[i]:
            bozo_q.pop()
            answer += 1
            continue
        
        while q and q[0] != order[i]:
            bozo_q.append(q.popleft())
            
        if q and q[0] == order[i]:
            q.popleft()
            answer += 1
        
        else:
            break
    
    return answer        